spaces.go

```diff
@@ -0,0 +1,413 @@
+	t, err := time.Parse(format, date)
+	if err != nil {
+		panic(fmt.Sprintf("Error in spaces.go while parsing %s as a JSON date", date))
+	}
+	return t
+}
+
+// ParseDDMMYYDate parses user-inputted dd/mm/yy date into time.Time
+func ParseDDMMYYDate(date string) (time.Time, error) {
+	//Attempt to parse as dd/mm/yy
+	format := "02/01/06"
+	t, err := time.Parse(format, date)
+	if err != nil {
+		// Attempt to parse as dd/m/yy
+		format = "02/1/06"
+		t, err = time.Parse(format, date)
+	}
+	if err != nil {
+		// Attempt to parse as d/mm/yy
+		format = "2/01/06"
+		t, err = time.Parse(format, date)
+	}
+	if err != nil {
+		// Attempt to parse as d/m/yy
+		format = "2/1/06"
+		t, err = time.Parse(format, date)
+	}
+	if err != nil {
+		// Attempt to parse as some form of dd/mm
+		// Attempt to parse as dd/mm
+		format = "02/01"
+		t, err = time.Parse(format, date)
+		if err != nil {
+			// Attempt to parse as dd/m
+			format = "02/1"
+			t, err = time.Parse(format, date)
+		}
+		if err != nil {
+			// Attempt to parse as d/mm
+			format = "2/01"
+			t, err = time.Parse(format, date)
+		}
+		if err != nil {
+			// Attempt to parse as d/m
+			format = "2/1"
+			t, err = time.Parse(format, date)
+		}
+
+		// Check if one of the dd/mm checks have worked
+		if err == nil {
+			// return t, but using the current year
+			year := time.Now().Year()
+			t = time.Date(year, t.Month(), t.Day(), 0, 0, 0, 0, t.Location())
+		}
+	}
+	return t, err
+}
+
+// FormatDate formats a time.Time into date in a standardised format
+func FormatDate(t time.Time) string {
+	return t.Format("Mon, 02 Jan 2006")
+}
+
+// FormatTime formats a time.Time into a time in a standardised format
+func FormatTime(t time.Time) string {
+	return t.Format("03:04PM")
+}
+
+// FormatTimeDate formats a time.Time into a full time and date, in a standardised format
+func FormatTimeDate(t time.Time) string {
+	return fmt.Sprintf("%s, %s", FormatTime(t), FormatDate(t))
+}
+
+// TimeInfo returns the time info of an Event in a readable format with duplicate info minimised
+func (event Event) TimeInfo() string {
+	// Ignores the case of "All Day" events, because there are none in database and I don't know how they're represented
+	// If the start and end of an event is on the same day, then the date is only stated once, with both start and
+	//   end times.
+	// If the start and end of an event is on different days, then both dates and both times are stated
+	start := ParseJSONDate(event.Start)
+	end := ParseJSONDate(event.End)
+	y1, m1, d1 := start.Date()
+	y2, m2, d2 := end.Date()
+
+	if y1 == y2 && m1 == m2 && d1 == d2 {
+		return fmt.Sprintf("%s to %s, %s", FormatTime(start), FormatTime(end), FormatDate(start))
+	}
+	return fmt.Sprintf("%s to %s", FormatTimeDate(start), FormatTimeDate(end))
+}
+
+// SpaceName returns the (capitalised) name of each Space. Case 4 (Prototyping Studio) unused and omitted.
+func SpaceName(facilityID int) string {
+	switch facilityID {
+	case 0:
+		return "THEME ROOM 1"
```

Comment:
String constants should be declared as constants. 

A simple guide here: https://tour.golang.org/basics/15. And ideally, we can declare them above in the `const` section of the file.